### ISSUE TITLE
[3.5.23] - [id] wildcard matched higher than explicit selector

### ISSUE THREAD

**Issue by austin43** [expo/expo#31800]
### Minimal reproducible example

https://github.com/austin43/expo-router-bug-3.5.21

### Which package manager are you using? (Yarn is recommended)

bun

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

metro

### Summary

After installing 3.5.21, I noticed our explicit routes were no longer being matched over [id] wildcard routes. When we routed to "/nested/create" for example, it would render the "/nested/[id]" route with create as the id param.

Click either button on the homepage of example app, and notice how each goes to the [id] page, rather than one going to [id] and the other to create.

This issue was then supposed to be fixed by https://github.com/expo/expo/pull/30909, but I think it's still a problem.

Original issue: https://github.com/expo/expo/issues/30879

### Environment

```text
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.6.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.16.0 - ~/.volta/tools/image/node/20.16.0/bin/node
      Yarn: 1.22.22 - ~/.bun/bin/yarn
      npm: 10.8.1 - ~/.volta/tools/image/node/20.16.0/bin/npm
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      Android SDK:
        API Levels: 28, 31
        Build Tools: 30.0.2, 30.0.3, 31.0.0
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: /undefined - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.25 => 51.0.25 
      expo-router: ^3.5.23 => 3.5.23 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.5 => 0.74.5 
      react-native-web: ~0.19.10 => 0.19.12 
    Expo Workflow: managed
```

**Comment by timoisalive** [comment]
Should the #30909 be available in 3.5.23? Changelog indicates it is unpublished...

**Comment by austin43** [comment]
@timoisalive I believe you are correct. Last I had checked a week or two ago the change log was not published but latest was pointed to 3.5.23. Closing ticket and will reopen if forthcoming release does not fix this.
